libs/sdk-core/src/backup.rs

```diff
@@ -63,8 +64,8 @@ pub struct BackupState {
 /// BackupTransport is the interface for syncing the sdk state between multiple apps.
 #[tonic::async_trait]
 pub trait BackupTransport: Send + Sync {
-    async fn pull(&self) -> Result<Option<BackupState>>;
-    async fn push(&self, version: Option<u64>, data: Vec<u8>) -> Result<u64>;
+    async fn pull(&self) -> NodeResult<Option<BackupState>>;
+    async fn push(&self, version: Option<u64>, data: Vec<u8>) -> NodeResult<u64>;
```

Comment:
This trait should be detached from the node api. We do implement it using greenlight today but it can be easily extended to use external service (Gdrive, icloud, remote server, etc...).
My main comment here is that I think it shouldn't return NodeResult. It seems to me that backup module should have its own errors (or Generic).